openapi.yaml

```diff
@@ -720,3 +837,61 @@ components:
           title: Current page number
           type: string
           example: "1"
+    CreateAclRequest:
```

Comment:
This has significantly less properties than the API specs that would be generated from the Java models, because I cut out everything from the `AclRequestsModel` that is not used by the `post` request. 

As I see it, if the primary goal of this openapi spec is to generate TS types, then we should try to be as accurate and focused as possible for clarity and dev experience. 

However, it means that we have to representation of the API definitions, one for the FE and one for the BE. Is that desirable?

@muralibasani brought this up this morning.

Reply:
Our goal is to have one representation of the API model. This could, long term, enable us to do automated testing so avoid for example changes to the API in the backend breaking things without us noticing immediately. 

We use the `/domain` directory and the exported types there to separate a bit better what our FE app needs to know from the backend. Types exported in `/domain` that are used in `/app` can only contain the properties we need and have a structure we need. 

If you see `topic-types`, currently we have `type Topic = KlawApiModel<"TopicInfo">` 
But `Topic` could look different from `KlawApiModel<"TopicInfo">`, eg. have properties removed / transformed etc. 

In that case, our `/app` would never know how `KlawApiModel<"TopicInfo">` looks like, while in `/domain` we still have the representation of the exact model from the backend. 

Would that fit what you want to do?

Reply:
Ah yes, I completely forgot about how `domain` fits in all this! It is the place where we do what I'm trying to do in the openapi defintions.

So the openapi specs should absolutely *not* do what I described in my comment haha! Will add the deleted properties back in.

Reply:
But then, we should have only one `AclRequest` component, and not one per type of request. Because I think @muralibasani explained that the is only one `AclRequest` entity that is used by all endpoints having to do with the entity.